### DISCUSSION TITLE
Make Group an interface

### DISCUSSION THREAD

**Discussion by wesen** [charmbracelet/huh#565] [Ideas]
I want to experiment with different Group layouts (for example as rows, as many of my forms are short fields, but many fields). Currently, I can't really do too much because I have to use the Group struct. I would like to play with the group titles and such as well.

Turning Group into an interface would make it fairly easy to experiment more (like with Field). This would also allow pulling in thirdparty huh-improvements easier.

I'm happy to do the necessary work, but wanted to check in before opening a pull request.

**Comment by maaslalani** [comment]
We'd be open to supporting different layouts such as horizontal layouts. We can likely introduce an option on a `Group` called `WithLayout` which perhaps can take `LayoutHorizontal` and `LayoutVertical` and displays the view accordingly.

**Comment by bashbunni** [comment]
Hey @wesen, thank you so much for wanting to make this project better! I've converted this to a discussion so we can talk through it and keep our backlog clear. If it does end up being something we want to add, we'll promote it back to an issue when we are ready to work on it.

If we commit to this feature, that's a feature we will need to maintain for the lifespan of this project. So, since "Yes is forever", we are carefully considering new features.

My personal thoughts: there is already a lot of abstraction going on in this codebase. If we were to change the design of Huh, we would want to carefully consider what that changes mean for users. On that note, if you have an example on what this would look like for users, that would really help us see your vision.
